hibernate-core/src/main/java/org/hibernate/id/uuid/UuidVersion7Strategy.java

```diff
@@ -101,21 +99,17 @@ public UUID generateUuid(SharedSessionContractImplementor session) {
 		);
```

Comment:
I just found this section which says that the least significant bit must be set:

>Alternatively, implementations MAY elect to obtain a 48-bit cryptographic-quality random number as per [Section 6.9](https://datatracker.ietf.org/doc/html/rfc9562#unguessability) to use as the Node ID. After generating the 48-bit fully randomized node value, implementations MUST set the least significant bit of the first octet of the Node ID to 1. This bit is the unicast or multicast bit, which will never be set in IEEE 802 addresses obtained from network cards. Hence, there can never be a conflict between UUIDs generated by machines with and without network cards. An example of generating a randomized 48-bit node value and the subsequent bit modification is detailed in [Appendix A](https://datatracker.ietf.org/doc/html/rfc9562#test_vectors). For more information about IEEE 802 address and the unicast or multicast or local/global bits, please review [[RFC9542](https://datatracker.ietf.org/doc/html/rfc9542)].

So you should probably add `| 1L` to the second argument.

Reply:
This is LSB of first octet, so this should be ` | 0x1000_0000_0000L `